### PR TITLE
Utf8 fixes

### PR DESCRIPTION
**Pull request by markeel**
This changes the encoding of any string from libgit2 to be treated as UTF-8.  The only down side of this fix is if someone was already using something like ASCII-8BIT, but that seems less likely than someone using UTF-8 but with characters that are multi-byte.

- *Production edit: This closes https://github.com/godotengine/godot-git-plugin/issues/150.*

### CODE REVIEW

**Comment by markeel** [comment]
I submitted a pull request a year ago, and it still has to be reviewed.  I basically forgot about it (and have been using my version of this plugin since then), How do we move this along?

**Comment by Calinou** [comment]
> I submitted a pull request a year ago, and it still has to be reviewed. I basically forgot about it (and have been using my version of this plugin since then), How do we move this along?

Please rebase this against the latest `master` branch so this can be merged :slightly_smiling_face: 

See [Pull request workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

File: .gitmodules [hunk 1]
```diff
@@ -5,6 +5,7 @@
 [submodule "libgit2"]
 	path = thirdparty/git2/libgit2
 	url = https://github.com/libgit2/libgit2
+	ignore = untracked
```
Comment (Calinou) [inline .gitmodules:8]: Out of curiosity, why is this change needed?

Comment (markeel) [reply .gitmodules:8]: This was a year ago, I think adding that stopped git from telling me there were changes in the libgit2.  When I rebase against the latest master, I'll see if it was really necessary or just an artifact of me trying to figure out the problem.
